runtime/dynlink.c

```diff
@@ -229,6 +230,55 @@ void caml_free_shared_libs(void)
     caml_dlclose(shared_libs.contents[--shared_libs.size]);
 }
 
+static value list_of_ext_table(struct ext_table* tbl)
+{
+  CAMLparam0();
+  CAMLlocal3(list, tmp, str);
+  int i;
+  for (i = tbl->size - 1; i >= 0; i--) {
+    str = caml_copy_string(tbl->contents[i]);
+    tmp = caml_alloc_small(2, 0);
+    Field(tmp, 0) = str;
+    Field(tmp, 1) = list;
+    list = tmp;
+  }
+  CAMLreturn (list);
+}
+
+CAMLprim value caml_dynlink_get_bytecode_sections(value unit)
+{
+  CAMLparam1(unit);
+  CAMLlocal2(ret, tbl);
+  ret = caml_alloc(4, 0);
+
+  if (caml_params->section_table != NULL) {
+    const char* sec_names[] = {"SYMB", "CRCS", "PRIM", "DLPT"};
+    int i, j;
+    tbl = caml_input_value_from_block(caml_params->section_table,
+                                      caml_params->section_table_size);
+    for (i = 0; i < sizeof(sec_names)/sizeof(sec_names[0]); i++) {
+      for (j = 0; j < Wosize_val(tbl); j++) {
+        value kv = Field(tbl, j);
+        if (!strcmp(sec_names[i], String_val(Field(kv, 0))))
+          Store_field(ret, i, Field(kv, 1));
+      }
+    }
+  } else {
+    Store_field(ret, 0,
+      caml_input_value_from_block(symb_section, symb_section_len));
+    if (crcs_section != NULL) {
+      Store_field(ret, 1,
+        caml_input_value_from_block(crcs_section, crcs_section_len));
+    }
+    Store_field(ret, 2,
+      list_of_ext_table(&caml_prim_name_table));
```

Comment:
If the `PRIM` section is removed from `-custom` executables (as suggested in `bytelink.ml`, this becomes common to both branches (and there's no `"PRIM"` section above)